source/2_platform/rex_directx/src/directx_feature_level.cpp

```diff
@@ -0,0 +1,49 @@
+#include "rex_directx/directx_feature_level.h"
+
+#include "rex_directx/directx_util.h"
+#include "rex_directx/dxgi/util.h"
+#include "rex_engine/diagnostics/assert.h"
+#include "rex_std/array.h"
+
+namespace rex
+{
+  const rsl::array g_expected_feature_levels = {D3D_FEATURE_LEVEL::D3D_FEATURE_LEVEL_12_0, D3D_FEATURE_LEVEL::D3D_FEATURE_LEVEL_12_1};
+
+  //-------------------------------------------------------------------------
+  rsl::string feature_level_name(const D3D_FEATURE_LEVEL& level)
```

Comment:
this can return a stack string and can use the enum reflection to avoid writing the function in the first place